docs/Installation-Script/UnityStationInstaller.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# Fonction de gestion des erreurs
+function handle_error {
+    echo "Une erreur s'est produite. Sortie du script."
+    exit 1
+}
+
+# Fonction pour obtenir la dernière version depuis l'API GitHub
+function get_latest_version {
+    local repo_url="https://api.github.com/repos/unitystation/stationhub/releases/latest"
+    local version=$(wget -qO - "$repo_url" | grep -oP '"tag_name": "\K(.*)(?=")')
+    echo "$version"
+}
+
+latest_version=$(get_latest_version)
+
+while true; do
+    echo "Choose your distribution :"
+    echo "1 - Ubuntu/Debian"
+    echo "2 - Arch-Linux"
+    echo "3 - Fedora"
+    echo "4 - OpenSUSE"
+    echo "5 - CentOS"
+    echo "6 - Gentoo"
+    read -p "Enter the number of your choice : " choice
+
+    case $choice in
+        1)
+            # Commandes pour Ubuntu
+            apt update
+            apt full-upgrade
+            apt install -y wget p7zip || handle_error
+            ;;
+
+        2)
+            # Commandes pour Arch Linux
+            pacman -Syu --noconfirm
+            pacman -S --noconfirm wget p7zip || handle_error
+            ;;
+
+        3)
+            # Commandes pour Fedora
+            dnf update -y
+            dnf install -y wget p7zip || handle_error
+            ;;
+
+        4)
+            # Commandes pour openSUSE
+            zypper refresh
+            zypper install -y wget p7zip || handle_error
+            ;;
+
+        5)
+            # Commandes pour CentOS
+            yum update -y
+            yum install -y wget p7zip || handle_error
+            ;;
+
+        6)
+            # Commandes pour Gentoo
+            emerge --sync
+            emerge -av wget p7zip || handle_error
+            ;;
+
+        *)
+            echo "Invalid choice. Please try again."
+            ;;
+    esac
+
+    if [ $choice -ge 1 ] && [ $choice -le 6 ]; then
+        break  # Sort de la boucle si le choix est valide
+    fi
+done
+
+# Téléchargement des fichiers avec wget
+mkdir -p /usr/share/Unitystation/
+cd /usr/share/Unitystation/
+latest_url="https://github.com/unitystation/stationhub/releases/download/$latest_version/lin$latest_version.zip"
+wget -O "/usr/share/Unitystation/lin-latest.zip" "$latest_url"
+wget -O /usr/share/Unitystation/unityico.png https://raw.githubusercontent.com/unitystation/stationhub/develop/UnitystationLauncher/Assets/unityico.png || handle_error
+wget -O /usr/share/applications/Stationhub.desktop https://raw.githubusercontent.com/Unitystation-fork/Unitystation-Others/main/Installation-Script/Unitystation.desktop || handle_error
+
+# Extraction du fichier zip
+unzip /usr/share/Unitystation/lin-latest.zip || handle_error
```

Comment:
looks like you are using ``unzip`` here, so what do you need ``p7zip`` for?

Reply:
of memory when you want to use archive extration via a script (for example sh or py) you need p7zip

but I could be mistaken

Reply:
`7zip` and `unzip` are both valid, some distros for some reason dont' come with 7zip

Reply:
in that case, the script would also have to find if the system has either and use that, like I suggested for curl and wget

Reply:
also you can do this to check for the existence of 7zip:

```sh
echo "Looking for 7zip.."
if ! command -v 7z &> /dev/null
then
    echo "7zip could not be found. Extract files manually."
    case $OS in
    'Linux')
        xdg-open https://7-zip.org/
        ;;
    'Windows')
        msg "%username%" 7zip could not be found. Extract files manually.
        start https://7-zip.org/
        ;;
    'Mac') 
        open https://7-zip.org/
        ;;
    'AIX') ;;
    *) ;;
    esac
    exit 1
fi
```

`if ! command -v 7z &> /dev/null` works on all platforms and distros + it checks if `7z` is available, then it does things based on the user's OS; which can also be used to check if `unzip` is available as well and fall back to it.

Reply:
Your script is crap and you know it.
opening a terminal and then opening a web page, did you miss the lessons on intuitiveness and the 3-click rule or what? does na notion of "user-friendly" mean anything to you?

Reply:
You mad lol

Reply:
> You mad lol

**eventually,** 
but you did create a script because you wanted to do what I did, as if you wanted my approval and recognition, it's a shame not to have your own ideas and have to copy others because you have no talent.

In short, gilles and bod have given me recommendations as soon as I have a bit of time, and I'm going to redo it.

However, **I have to thank you,** so   you motivated me to improve my script to prove you wrong, =)